--- GConf-2.7.91.1/configure.in.orig	2004-08-19 13:17:00.000000000 +0200
+++ GConf-2.7.91.1/configure.in	2004-08-19 22:32:51.588746296 +0200
@@ -240,7 +240,7 @@
 
 AC_CHECK_FUNCS(flockfile)
 
-ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el en_CA en_GB es eu fa fi fr ga gl gu hi hr hu id it ja ko lt lv mk ml mn ms nb ne nl nn no pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta tr uk vi yi zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el en_CA en_GB es eu fa fi fr ga gl gu hi hr hu id it ja ko lt lv mk ml mn ms nb ne nl nn pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta tr uk vi yi zh_CN zh_TW"
 
 AM_GLIB_GNU_GETTEXT
 
