diff -Nur GConf-1.2.1/backends/Makefile.am GConf-1.2.1.new/backends/Makefile.am
--- GConf-1.2.1/backends/Makefile.am	Wed Jun 12 22:32:12 2002
+++ GConf-1.2.1.new/backends/Makefile.am	Thu Aug 15 02:19:08 2002
@@ -4,7 +4,7 @@
 
 EXTRA_DIST=README.bdb
 
-backenddir = $(pkglibdir)/$(MAJOR_VERSION)
+backenddir = $(pkglibdir)$(MAJOR_VERSION)
 
 if BDB
 BDBBACKEND=libgconfbackend-bdb.la
diff -Nur GConf-1.2.1/gconf/Makefile.am GConf-1.2.1.new/gconf/Makefile.am
--- GConf-1.2.1/gconf/Makefile.am	Tue Jul 23 19:14:07 2002
+++ GConf-1.2.1.new/gconf/Makefile.am	Thu Aug 15 02:21:59 2002
@@ -9,12 +9,12 @@
 	-DG_LOG_DOMAIN=\"GConf\"					\
 	-DGCONF_LOCALE_DIR=\""$(gconflocaledir)"\"			\
         -DGCONF_SRCDIR=\""$(absolute_top_srcdir)"\"			\
-	-DGCONF_CONFDIR=\""$(sysgconfdir)/$(MAJOR_VERSION)"\"		\
+	-DGCONF_CONFDIR=\""$(sysgconfdir)"\"		\
 	-DGCONF_ETCDIR=\""$(sysgconfdir)"\"				\
 	-DGCONF_BINDIR=\""$(bindir)"\"					\
 	-DGCONF_SERVERDIR=\""$(libexecdir)"\"           \
 	-DGCONF_BUILDDIR=\""$(top_builddir)"\"				\
-	-DGCONF_BACKEND_DIR=\""$(pkglibdir)/$(MAJOR_VERSION)"\"		\
+	-DGCONF_BACKEND_DIR=\""$(pkglibdir)$(MAJOR_VERSION)"\"				\
 	-DVERSION=\""$(VERSION)"\"					\
 	-DGCONF_ENABLE_INTERNALS=1					\
 	-DGCONFD=\""$(GCONFD_BINARY_NAME)"\"				\
@@ -63,7 +63,7 @@
 	gconf-engine.h		\
 	gconf-client.h
 
-gconfincludedir = $(includedir)/gconf/$(MAJOR_VERSION)/gconf
+gconfincludedir = $(includedir)/gconf$(MAJOR_VERSION)/gconf
 gconfinclude_HEADERS = 		\
 	$(gconf_headers) 	\
 	gconf-enum-types.h
@@ -122,8 +122,8 @@
 EXTRA_DIST=GConf.idl default.path.in gconfmarshal.list regenerate-enum-header.sh regenerate-enum-footer.sh
 
 install-data-local:
-	-mkdir -p $(DESTDIR)$(sysgconfdir)/$(MAJOR_VERSION)
-	$(INSTALL_DATA) default.path $(DESTDIR)$(sysgconfdir)/$(MAJOR_VERSION)/path
+	-mkdir -p $(DESTDIR)$(sysgconfdir)
+	$(INSTALL_DATA) default.path $(DESTDIR)$(sysgconfdir)/path
 
 gconfmarshal.h: @REBUILD@ stamp-gconfmarshal.h
 	@true
diff -Nur GConf-1.2.1/gconf-2.0.pc.in GConf-1.2.1.new/gconf-2.0.pc.in
--- GConf-1.2.1/gconf-2.0.pc.in	Tue Jul 23 19:14:07 2002
+++ GConf-1.2.1.new/gconf-2.0.pc.in	Thu Aug 15 02:19:08 2002
@@ -10,4 +10,4 @@
 Version: @VERSION@
 Requires: ORBit-2.0
 Libs: -L${libdir} -lgconf-@MAJOR_VERSION@
-Cflags: -I${includedir}/gconf/@MAJOR_VERSION@
+Cflags: -I${includedir}/gconf@MAJOR_VERSION@
