From 6160b99595275f9942bc90ae2067e0349889e589 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 20 May 2025 13:40:30 +0200
Subject: [PATCH] Use gettext 0.21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 configure.ac |   3 +++

diff --git a/configure.ac b/configure.ac
index 69a57e6..3c6ba2a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,11 +3,14 @@ AC_INIT([GConf],
   [3.2.6],
   [http://bugzilla.gnome.org/enter_bug.cgi?product=GConf],
   [GConf])
+AC_CONFIG_MACRO_DIRS([m4])
 
 AC_CONFIG_SRCDIR([gconf/gconf.h])
 AC_CONFIG_HEADERS([config.h])
 
 AM_INIT_AUTOMAKE([1.9 no-dist-gzip dist-bzip2 -Wall])
+AM_GNU_GETTEXT_VERSION(0.21)
+AM_GNU_GETTEXT([external])
 
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
-- 
2.49.0

