diff -urN GConf-2.6.1.orig/gconf/default.path.in GConf-2.6.1/gconf/default.path.in
--- GConf-2.6.1.orig/gconf/default.path.in	2004-04-19 23:32:25.705203536 +0200
+++ GConf-2.6.1/gconf/default.path.in	2004-04-19 23:34:51.734003784 +0200
@@ -14,6 +14,9 @@
 # wide sources in this file.
 include @EXPANDED_SYSCONFDIR@/gconf/2/local-mandatory.path
 
+# Uncomment line below if GConf 1 is installed
+# xml:readonly:/etc/X11/GNOME/gconf/gconf.xml.mandatory
+
 # Now see where users want us to look - basically the user can stick arbitrary 
 # sources in a ~/.gconf.path file and they're inserted here
 include "$(HOME)/.gconf.path"
@@ -28,3 +31,6 @@
 
 # Finally, look at the systemwide defaults
 xml:readonly:@EXPANDED_SYSCONFDIR@/gconf/gconf.xml.defaults
+
+# Uncomment line below if GConf 1 is installed
+# xml:readonly:/etc/X11/GNOME/gconf/gconf.xml.defaults
