diff -urN aa/GConf-2.7.3/gconf/gconftool.c GConf-2.7.3/gconf/gconftool.c
--- aa/GConf-2.7.3/gconf/gconftool.c	2004-07-04 17:15:23.000000000 +0200
+++ GConf-2.7.3/gconf/gconftool.c	2004-07-07 18:23:53.900489072 +0200
@@ -3381,7 +3381,7 @@
     {
       GError* error = NULL;
 
-      if (!gconf_engine_associate_schema(conf, tmp->data, unload ? schema_name : NULL, &error))
+      if (!gconf_engine_associate_schema(conf, tmp->data, !unload ? schema_name : NULL, &error))
         {
           g_assert(error != NULL);
           
