diff -urN GConf-1.2.1.orig/gconf/default.path.in GConf-1.2.1/gconf/default.path.in
--- GConf-1.2.1.orig/gconf/default.path.in	Fri Oct 18 23:46:43 2002
+++ GConf-1.2.1/gconf/default.path.in	Fri Oct 18 23:49:06 2002
@@ -9,6 +9,9 @@
 # (commented out until xml backend knows how to be read-only for users)
 xml:readonly:@EXPANDED_SYSCONFDIR@/gconf/gconf.xml.mandatory
 
+# Uncomment line below if GConf 1 is installed
+# xml:readonly:/etc/X11/GNOME/gconf/gconf.xml.mandatory
+
 # Now see where users want us to look - basically the user can stick arbitrary 
 # sources in a ~/.gconf.path file and they're inserted here
 include "$(HOME)/.gconf.path"
@@ -21,5 +24,6 @@
 # (commented out for now)
 xml:readonly:@EXPANDED_SYSCONFDIR@/gconf/gconf.xml.defaults
 
-
+# Uncomment line below if GConf 1 is installed
+# xml:readonly:/etc/X11/GNOME/gconf/gconf.xml.defaults
 
