--- GConf-1.1.8/backends/Makefile.am~	Wed Feb 20 21:05:23 2002
+++ GConf-1.1.8/backends/Makefile.am	Wed Feb 20 21:05:23 2002
@@ -4,7 +4,7 @@
 
 EXTRA_DIST=README.bdb
 
-backenddir = $(pkglibdir)/$(MAJOR_VERSION)
+backenddir = $(pkglibdir)$(MAJOR_VERSION)
 
 if BDB
 BDBBACKEND=libgconfbackend-bdb.la
--- GConf-1.1.8/gconf/Makefile.am~	Wed Feb 20 21:19:39 2002
+++ GConf-1.1.8/gconf/Makefile.am	Wed Feb 20 21:19:39 2002
@@ -9,11 +9,11 @@
 	-DG_LOG_DOMAIN=\"GConf\"					\
 	-DGCONF_LOCALE_DIR=\""$(gconflocaledir)"\"			\
         -DGCONF_SRCDIR=\""$(absolute_top_srcdir)"\"			\
-	-DGCONF_CONFDIR=\""$(sysconfdir)/gconf/$(MAJOR_VERSION)"\"	\
+	-DGCONF_CONFDIR=\""$(sysconfdir)/gconf/"\"	\
 	-DGCONF_ETCDIR=\""$(sysconfdir)/gconf"\"			\
 	-DGCONF_BINDIR=\""$(bindir)"\"					\
 	-DGCONF_BUILDDIR=\""$(top_builddir)"\"				\
-	-DGCONF_BACKEND_DIR=\""$(pkglibdir)/$(MAJOR_VERSION)"\"		\
+	-DGCONF_BACKEND_DIR=\""$(pkglibdir)$(MAJOR_VERSION)"\"		\
 	-DVERSION=\""$(VERSION)"\"					\
 	-DGCONF_ENABLE_INTERNALS=1					\
 	-DGCONFD=\""$(GCONFD_BINARY_NAME)"\"				\
@@ -61,7 +61,7 @@
 	gconf-engine.h		\
 	gconf-client.h
 
-gconfincludedir = $(includedir)/gconf/$(MAJOR_VERSION)/gconf
+gconfincludedir = $(includedir)/gconf$(MAJOR_VERSION)/gconf
 gconfinclude_HEADERS = 		\
 	$(gconf_headers) 	\
 	gconf-enum-types.h
@@ -120,8 +120,8 @@
 EXTRA_DIST=GConf.idl default.path.in gconfmarshal.list regenerate-enum-header.sh regenerate-enum-footer.sh
 
 install-data-local:
-	-mkdir -p $(DESTDIR)$(sysconfdir)/gconf/$(MAJOR_VERSION)
-	$(INSTALL_DATA) default.path $(DESTDIR)$(sysconfdir)/gconf/$(MAJOR_VERSION)/path
+	-mkdir -p $(DESTDIR)$(sysconfdir)/gconf/
+	$(INSTALL_DATA) default.path $(DESTDIR)$(sysconfdir)/gconf/path
 
 gconfmarshal.h: @REBUILD@ stamp-gconfmarshal.h
 	@true
--- GConf-1.1.8/gconf-2.0.pc.in~	Wed Feb 20 21:43:34 2002
+++ GConf-1.1.8/gconf-2.0.pc.in	Wed Feb 20 21:43:34 2002
@@ -9,4 +9,4 @@
 Version: @VERSION@
 Requires: ORBit-2.0
 Libs: -L${libdir} -lgconf-@MAJOR_VERSION@
-Cflags: -I${includedir}/gconf/@MAJOR_VERSION@
+Cflags: -I${includedir}/gconf@MAJOR_VERSION@
