--- GConf-2.4.0.1/configure.in.orig	Wed Sep 10 21:23:11 2003
+++ GConf-2.4.0.1/configure.in	Fri Feb 13 11:39:22 2004
@@ -216,7 +216,7 @@
 
 AC_CHECK_FUNCS(flockfile)
 
-ALL_LINGUAS="am ar az be bg bn ca cs cy da de el es eu fa fi fr ga gl hi hu id it ja ko lv mk ml mn ms nl nn no pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi yi zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn ca cs cy da de el es eu fa fi fr ga gl hi hu id it ja ko lv mk ml mn ms nl nn nb pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi yi zh_CN zh_TW"
 
 AM_GLIB_GNU_GETTEXT
 
